packages/node_modules/@webex/plugin-meetings/src/reachability/index.js

```diff
@@ -182,6 +181,58 @@ export default class Reachability extends StatelessWebexPlugin {
     });
   }
 
+
+  /**
+   * An event handler on an RTCPeerConnection when the state of the ICE
+   * candidate gathering process changes. Used to measure connection
+   * speed.
+   * @private
+   * @param {RTCPeerConnection} peerConnection
+   * @returns {Promise}
+   */
+  iceGatheringState = (peerConnection) =>
+    new Promise((resolve) => {
+      const waitTime = 1e4;
+      const decimalPoints = 6;
+      const getElapsedTime = () => ((window.performance.now() - peerConnection.begin) / 1000).toFixed(decimalPoints);
+
+      const timeout = setTimeout(() => {
+        LoggerProxy.logger.error(`Reachability:index#iceGatheringState --> No ice candidate for ${peerConnection.key}.`);
+        resolve({clusterId: peerConnection.key, elapsed: null});
+      }, waitTime);
+
+      const setLatencyAndClose = (elapsed) => {
+        peerConnection.onicecandidate = null;
+        peerConnection.elapsed = elapsed;
+        peerConnection.close();
+        clearTimeout(timeout);
+      };
+
+      // eslint-disable-next-line no-param-reassign
+      peerConnection.onicegatheringstatechange = () => {
+        const {COMPLETE} = ICE_GATHERING_STATE;
+
+        if (peerConnection.iceConnectionState === COMPLETE) {
+          const elapsed = getElapsedTime(peerConnection);
+
+          LoggerProxy.logger.info(`Reachability:index#onIceGatheringStateChange --> Successfully pinged ${peerConnection.key}:`, elapsed);
+          setLatencyAndClose(elapsed);
+          resolve({clusterId: peerConnection.key, elapsed});
+        }
+      };
+
+      peerConnection.onicecandidate = (e) => {
+        if (e.candidate) {
+          const elapsed = getElapsedTime(peerConnection);
```

Comment:
`getElapsedTime` doesn't accept any params, does the function need to be updated?

Reply:
The function is using closure to a reference the **peerConnection**, I could pass it but it would be redundant I feel. I tried to break the code into smaller functions and reduce that long code block that was within the **forEach** loop. I choose names for the functions that self document what's going on. 

This whole function ** iceGatheringState** applies to only one peerConnection within the closure.

Reply:
Ok, that works, should the code not pass in the parameter then like:

```suggestion
          const elapsed = getElapsedTime();
```

Reply:
Thank you for catching that Adam! I'll fix it. I was originally passing it until I decided it was redundant since I can just use the benefits of the closure.